code/modules/projectiles/guns/energy.dm

```diff
@@ -133,6 +132,10 @@
 	if(!automatic_charge_overlays)
 		return
 	var/ratio = CEILING((cell.charge / cell.maxcharge) * charge_sections, 1)
+	if(ratio == old_ratio)
+		return

```

Comment:
This means changes that do not update cell fail to happen ? For example update_icon after changing stun/kill modes.